addons/folo/common/src/main/java/org/commonjava/indy/folo/data/idxmodel/TrackingKey2StringMapper.java

```diff
@@ -40,11 +48,21 @@ public String getStringMapping( Object key )
             return tk.getId();
 
         }
+        else if ( key instanceof WrappedByteArray )
+        {
+            return generateString( BYTEARRAYKEY_IDENTIFIER,
```

Comment:
I think if we store the cache off-heap it will just serialize TrackingKey. If we're persisting that cache out of off-heap memory and down to the database, we'd need to deserialize from `byte[]` to TrackingKey (using `new ObjectInputStream(new ByteArrayInputStream(key.getBytes()))` to read the TrackingKey instance), then serialize to string the same way we would if we had the TrackingKey object in the first place. It just take the one-step process we already had (encode as string) and turns it into a two-step process (deserialize + encode).

Reply:
Fixed with ObjectInputStream way. And I think maybe we should implement a abstract class to normalize this type of stuff.

Reply:
I've also extracted a abstract class which handle this type of object from upper level. See AbstractIndyKey2StringMapper in new added commit.